server/main.go

```diff
@@ -80,6 +85,18 @@ type Main struct {
 	TreeDeleteMinInterval time.Duration
 }
 
+func (m *Main) healthz(rw http.ResponseWriter, req *http.Request) {
+	if m.IsHealthy != nil {
+		if err := m.IsHealthy(req.Context()); err != nil {
+			rw.WriteHeader(http.StatusServiceUnavailable)
+			rw.Write([]byte(err.Error()))
+			return
+		}
+	}
+	rw.WriteHeader(http.StatusOK)
```

Comment:
Pretty sure there's an implicit 200 status if not overridden.

Reply:
done